File: tests/testdata/codegen/unknown-resource-pp/nodejs/unknown-resource.ts
import * as codeinfra from "@codeinfra/codeinfra";
import * as unknown from "@codeinfra/unknown";

const provider = new codeinfra.providers.Unknown("provider", {});
const main = new unknown.index.Main("main", {
    first: "hello",
    second: {
        foo: "bar",
    },
});
const fromModule: unknown.eks.Example[] = [];
for (const range = {value: 0}; range.value < 10; range.value++) {
    fromModule.push(new unknown.eks.Example(`fromModule-${range.value}`, {associatedMain: main.id}));
}
export const mainId = main.id;
export const values = fromModule.values.first;

File: tests/testdata/codegen/unknown-resource-pp/python/unknown-resource.py
import codeinfra
import codeinfra_unknown as unknown

provider = codeinfra.providers.Unknown("provider")
main = unknown.index.Main("main",
    first=hello,
    second={
        foo: bar,
    })
from_module = []
for range in [{"value": i} for i in range(0, 10)]:
    from_module.append(unknown.eks.Example(f"fromModule-{range['value']}", associated_main=main.id))
codeinfra.export("mainId", main["id"])
codeinfra.export("values", from_module["values"]["first"])
